pxtblocks/fields/field_melodySandbox.ts

```diff
@@ -0,0 +1,415 @@
+/// <reference path="../../built/pxtlib.d.ts" />
+
+namespace pxtblockly {
+    export class FieldCustomMelody<U extends Blockly.FieldCustomOptions> extends Blockly.Field implements Blockly.FieldCustom {
+        public isFieldCustom_ = true;
+        protected params: U;
+        private title: string = "Name this tune";
+        private melody: pxtmelody.MelodyArray;
+        private soundingKeys: number = 0;
+        private numRow: number = 8;
+        private numCol: number = 8;
+        private tempo: number = 120;
+        private stringRep: string;
+        private oneNotePerCol: boolean = true;
+        private isPlaying: boolean = false;
+        private isLooping: boolean = false;
+        private timeouts: any = []; // keep track of timeouts
+        private duration = 60000 / this.tempo; //ms to hold note
+
+        constructor(value: string, params: U, validator?: Function) {
+            super(value, validator);
+            this.params = params;
+            if (!this.melody) {
+                this.melody = new pxtmelody.MelodyArray();
+            }
+        }
+
+        init() {
+            super.init();
+            this.onInit();
+        }
+
+        showEditor_() {
+            // If there is an existing drop-down someone else owns, hide it immediately and clear it.
+            Blockly.DropDownDiv.hideWithoutAnimation();
+            Blockly.DropDownDiv.clearContent();
+            Blockly.DropDownDiv.setColour(this.getDropdownBackgroundColour(), this.getDropdownBorderColour());
+            let contentDiv = Blockly.DropDownDiv.getContentDiv() as HTMLDivElement;
+            goog.style.setStyle(contentDiv, "max-height", "500px");
```

Comment:
Don't use `goog.style`

Reply:
@riknoll What's the issue with all the goog stuff? Does that use a lot more resources?

Reply:
Same reason as I gave for goog.dom, the Blockly team is planning on removing it. They want to get rid of the dependency.